pubsub/aws/snssqs/snssqs.go

```diff
@@ -243,48 +228,59 @@ func (s *snsSqs) getTopicArn(parentCtx context.Context, topic string) (string, e
 	})
 	cancelFn()
 	if err != nil {
-		return "", fmt.Errorf("error: %w while getting topic: %v with arn: %v", err, topic, arn)
+		return "", fmt.Errorf("error: %w, while getting (sanitized) topic: %v with arn: %v", err, topic, arn)
 	}
 
 	return *getTopicOutput.Attributes["TopicArn"], nil
 }
 
 // get the topic ARN from the topics map. If it doesn't exist in the map, try to fetch it from AWS, if it doesn't exist
 // at all, issue a request to create the topic.
-func (s *snsSqs) getOrCreateTopic(ctx context.Context, topic string) (topicArn string, sanitizedName string, err error) {
-	s.topicsLock.Lock()
-	defer s.topicsLock.Unlock()
+func (s *snsSqs) getOrCreateTopic(ctx context.Context, topic string) (string, string, error) {
+	var (
+		topicArn       string
+		loadOK         bool
+		sanitizedTopic string
+		err            error
+	)
```

Comment:
I do really like named returns when there's > 2 returned values and have the smae type. The signature of the method should be clear as to what values are returned.

There's no difference in adding the names or initializing the variables right after, but the former approach is clearer for readers.

(Named returns don't require "naked returns", you can still return the variables by name)

```suggestion
func (s *snsSqs) getOrCreateTopic(ctx context.Context, topic string) (topicArn string, sanitizedTopic string, err error) {
	var loadOK bool
```